nixos/modules/system/activation/top-level.nix

```diff
@@ -61,11 +61,17 @@ let
 
   # Handle assertions and warnings
 
-  failedAssertions = map (x: x.message) (filter (x: !x.assertion) config.assertions);
+  failedAssertions = let
+    assertions = lib.collect' (x: builtins.isAttrs x && x ? message && !(x.assertion or true) && (x.enable or false)) config.assertions;
+  in map (x: "assertions." + (lib.concatStringsSep "." x.path) + ":\n" + x.value.message) assertions;
+
+  failedWarnings = let
+    warnings = lib.collect' (x: builtins.isAttrs x && x ? message && (x.condition or false) && (x.enable or false)) config.warnings;
+  in map (x: "warnings." (lib.concatStringsSep "." x.path) + ":\n" + x.value.message) warnings;
```

Comment:
These have a bit of a problem at the moment. For a path like `[ "services" "asdf" "thing with spaces" "settings" ]`, it will print `services.asdf.thing with spaces.settings`, while it optimally should've been `services.asdf."thing with spaces".settings`. I'm not sure if it would be reasonable to add a function `showPath :: [String] -> String` to either `lib.strings` or `lib.attrsets`, or maybe we have something like it already?

Reply:
I've added a function to deal with this. Could discuss whether it should be part of `lib` at a later time.

Reply:
Sorry I didn't catch this comment earlier. We have `lib.showOption` for this.